shared-libs/transitions/src/transitions/index.js

```diff
@@ -161,8 +57,9 @@ const loadTransitions = () => {
     const conf = transitionsConfig[transition];
 
     const disabled = conf && conf.disable;
+    const allowSync = conf && !conf.async;
 
-    if (!conf || disabled) {
+    if (!conf || disabled || (!allowSync && synchronous)) {
```

Comment:
I chose to make the config "disable" some transitions from running synchronously instead of "enabling" to run synchronously (after back and fourth-ing a few times). Reasoning was to enable projects to start using this feature without changes to configurations.
AFAICT the only transitions which should run async are muting and update_notifications, which may end up needing to updated lots of documents.

I would appreciate some feedback on this.